components/places/src/bookmark_sync/store.rs

```diff
@@ -504,6 +512,50 @@ impl<'a> BookmarksStore<'a> {
         Ok(())
     }
 
+    /// Fetches a list of items that have been reuploaded with new structure for
+    /// the last 5 syncs. This is captured in logs and telemetry, and suggests
+    /// that we might be in a sync loop with another client (bug 1530145).
+    fn consecutive_reuploads(&self) -> Result<Vec<ConsecutiveReupload>> {
+        let mut reuploads = Vec::new();
+        let mut stmt = self.db.prepare(&format!(
+            "WITH
+             ranks(rank, at, guid, action) AS (
```

Comment:
/me googles `sqlite rank`.
/me goes "Nice!"

IIUC, this will rely on a single timestamp for all entries written here, which I believe you do correctly in store.rs, but I wonder if a comment there indicating how important the single timestamp is is worthwhile?

Reply:
It definitely is. I'd also like to test this out for large logs—what happens if we have 10k events? 100k?